src/lib/icli/index.js

```diff
@@ -0,0 +1,357 @@
+  parameter.name = _.camelCase(flags.shift());
+  return {
+    flags: parameter.cmdSpec,
+    description: parameter.description,
+    coercion: parameter.coercion || getCoercionForType(parameter.type)
+  };
+}
+
+/**
+ * Parse a parameter of type "argument" to calculate the name and generate the commander option
+ * @param  {Object} parameter - a parameter definition that will be enriched with a correct name
+ * @return {Object} - a commander argument
+ */
+function parseArgumentSpec(parameter) {
+  parameter.name = _.camelCase(parameter.cmdSpec);
+  return parameter.cmdSpec;
+}
+
+/**
+ * Get default coercion for a parameter type
+ * @param  {string} type - a parameter type
+ * @return {function|null} - the coercion function to applu to the command line argument/option
+ */
+function getCoercionForType(type) {
+  switch (type) {
+  case 'int':
+    return parseInt;
+  case 'checkbox':
+    return val => { return _.map(val.split(','), _.trim); };
+  default:
+    return null;
+  }
+}
+
+/**
+ * Construct the action() function passed to commander
+ * @param  {Array} parameters - the list of parameters
+ * @param  {function} executeCommand - the function that perform the interactive command task with property values
+ * @param  {function} commanderActionHook - a hook fuction that allows to alter the result of the command arguments/options
+ * @param  {function} inquirerPromptHook - a hook fuction that allows to alter the result of the questions
+ * @return {function} - The function that must be passed to cmd.action()
+ */
+function getAction(parameters, executeCommand, commanderActionHook, inquirerPromptHook) {
+  return function action() {
+    // Hook that allows to tranform the result of the commander parsing, before converting it into parameter values
+    const args = commanderActionHook ? commanderActionHook.apply(this, arguments) : arguments;
+    const validations = _.reduce(parameters, (result, parameter) => {
+      if (parameter.validate) {
+        result[parameter.name] = parameter.validate;
+      }
+      return result;
+    }, {});
+    const commandParameterValues = processCliArgs(args, validations);
+
+    // If the cli arguments are correct, we can prepare the questions for the interactive prompt
+    // Launch the interactive prompt
+    return prompt.prompt(parametersToQuestions(parameters, commandParameterValues))
+    .then(answers => {
+      // Hook that allows to tranform the result of the inquirer prompt, before converting it into parameter values
+      if (inquirerPromptHook) {
+        return inquirerPromptHook(answers, commandParameterValues);
+      }
+      // Once we have all parameter values from the command and from the questions, we can execute the command
+      _.forEach(answers, (value, key) => {
+        if (!commandParameterValues[key] && answers[key]) {
+          commandParameterValues[key] = answers[key];
+        }
+      });
+      executeCommand(commandParameterValues);
+    });
+  };
+}
+
+/**
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Missing JSDoc @returns for function.](https://www.codacy.com/app/lager/lager/file/2779497532/issues/source?bid=3360491&fileBranchId=3365065#l245)